heron/metricscachemgr/tests/java/com/twitter/heron/metricscachemgr/metricscache/CacheCoreTest.java

```diff
@@ -331,7 +329,13 @@ public void testComponent() {
         new MetricDatum("c2", "i2", "m1", Arrays.asList(
             new MetricTimeRangeValue(now - 90 * 1000, now - 90 * 1000, "0.7")
         ))
-    );
+    };
+    List<MetricDatum> metricList = new ArrayList<>(response.getMetricList());
+    assertEquals(metricList.size(), expected.length);
+    // sort
+    metricList.sort(datumComparator);
+    // there should be 4 component-instance
+    assertMetricResponse(metricList, expected);
```

Comment:
Every test does the following before calling `assertMetricResponse(metricList, expected)`:
```
  assertEquals(metricList.size(), expected.length);
  metricList.sort(datumComparator);
```
Why not move those inside the `assertMetricResponse` method?